src/lib/coerce.js

```diff
@@ -424,6 +424,22 @@ exports.coerceFont = function(coerce, attr, dfltObj) {
     return out;
 };
 
+/*
+ * Shortcut to coerce the pattern attributes
+ */
+exports.coercePattern = function(coerce, attr) {
+    var out = {};
+
+    out.shape = coerce(attr + '.shape');
+    if(out.shape) {
+        out.size = coerce(attr + '.size');
+        out.bgcolor = coerce(attr + '.bgcolor');
+        out.solidity = coerce(attr + '.solidity');
+    }
+
+    return out;
```

Comment:
Good idea to move this to `lib/coerce` for reuse! But I'd skip creating `out` unless it's actually going to be used.

Reply:
Done in https://github.com/plotly/plotly.js/pull/5537/commits/4de929d638aa9b0c4cb6e007a471508a69823a37.